### PR TITLE
Task Grid (Copy of PR #26, originally merged into the v2 branch).

### PR DESCRIPTION
**Pull request by Bonkles**
* Grid has been added if a GPX url is supplied- Grid options (2x2, 3x3, etc) can be found in the background controls if and only if the bounding area is a rectangle.
*ES6-ified the rapid_context module.

### CODE REVIEW

**Comment by Bonkles** [comment]
image
 Screenshot of v3 functionality:

File: modules/core/rapid_context.js [hunk 1]
```diff
@@ -22,13 +35,55 @@ export function coreRapidContext(context) {
                     if (maxlat === undefined || lat > maxlat) maxlat = lat;
                     if (maxlon === undefined || lon > maxlon) maxlon = lon;
                 }
+
+                if (f.geometry.type === 'LineString' && lineStringCount === 1) {
+                    var lats = f.geometry.coordinates.map((f) => {return f[0];});
```
Comment (bhousel) [inline modules/core/rapid_context.js:40]: These anonymous fat arrow functions can be written a bit more concisely..
With braces you can have multiple statements but must supply an explicit return.. Without braces, you have a single expression with an implicit return. 

```suggestion
                    var lats = f.geometry.coordinates.map(f => f[0]);
```

Comment (Bonkles) [reply modules/core/rapid_context.js]: Ooh, good catch. Also fixed the not-quite-es6ified filter & map calls on lines 49 and 51.

**Comment by Bonkles** [comment]
@bhousel can you take another look? I have since made a couple updates. 

BTW, the conflicts in the dist/locales files were resolved by my changes, not introduced, so I think we're good there? Let me know!

**Comment by bhousel** [comment]
> BTW, the conflicts in the dist/locales files were resolved by my changes, not introduced, so I think we're good there? Let me know!

Looks good!  I'll merge it 👍

**Pull request merged**